src/modules/user/user.usecase.spec.ts

```diff
@@ -217,6 +248,244 @@ describe('User use cases', () => {
       });
     });
   });
+
+  describe('changeUserEmailById', () => {
+    it('should change the user email successfully', async () => {
+      jest.spyOn(bridgeService, 'updateUserEmail').mockResolvedValue(undefined);
+      jest
+        .spyOn(sharedWorkspaceRepository, 'updateGuestEmail')
+        .mockResolvedValue(undefined);
+      jest.spyOn(userRepository, 'updateBy').mockResolvedValue(undefined);
```

Comment:
Mocking things like "`updateBy`" and "`updateById`" just to return undefined will make test files grow larger than needed. Try to only Spy on objects that you really need to return `undefined` to make tests easier to read.  

These tests run fine without spying on every "update xxx" call. Only spy on functions that you either require to check how it has been called or are essential. I know `findByUsername` and other `get functions` need to be mocked to undefined because by default it returns` {}` which is considered as an existent object so don't worry about them.